src/printer/index.js

```diff
@@ -302,7 +302,10 @@ function genericPrint(path, options, print) {
 
     case "float":
     case "int": {
-      return `${n.n}`;
+      const parent = path.getParentNode();
+      // This is overly cautious, we may  want to revisit and normalize more
+      // cases here.
+      return /[ex.]/i.test(parent.source) ? parent.source : `${n.n}`;
```

Comment:
there is probably no reason to ever normalize these. consider e.g. PEP 515 section dividers, plus in general you can't represent all python numbers exactly with the JS number type anyway

Reply:
Changed it to always return `parent.source`

Reply:
probably just apply this check for `Num` then, and don't bother with the special-case number handling

e.g. you can't differentiate `1.0j` from `1j` in the AST anyway, but those should not be reprinted

Reply:
i mean in `astexport` as well